boot/initramfs.go

```diff
@@ -0,0 +1,87 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package boot
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/seed"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/timings"
+)
+
+// TODO:UC20: add install mode mounts code too
+
+// InitramfsRunModeSnapsToMount returns a map of the snap paths to mount for the
+// specified snap types.
+func InitramfsRunModeSnapsToMount(typs []snap.Type) (map[snap.Type]string, error) {
+	var choices bootStateUpdate
+	var sn snap.PlaceInfo
+	var err error
+	m := make(map[snap.Type]string)
+	for _, typ := range typs {
+		ebs := newEarlyBootState20(typ)
```

Comment:
we don't have UC 16/18 variants of this, it's a bit unclear whether using the interface to write the code is useful, or just having a switch here is simpler, and use *State types more concretely